### PR TITLE
Add compatibility shim for endian related functions

### PR DESCRIPTION
**Pull request by oerdnj**
The <endian.h> isn't POSIX, add a compatibility shim (taken and
relicensed from BIND 9) to support various BSDs and macOS systems.

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -5,8 +5,9 @@ lib_LTLIBRARIES = pkcs11_provider.la
 SHARED_EXT=@SHARED_EXT@
 
 pkcs11_provider_la_SOURCES = \
-	asymmetric_cipher.c \
+asymmetric_cipher.c \
```
Comment (simo5) [inline src/Makefile.am:8]: why this change?

File: src/endian.h [hunk 2]
```diff
@@ -0,0 +1,155 @@
+/* Copyright (C) Internet Systems Consortium, Inc. ("ISC") - see COPYING */
+
+#pragma once
```
Comment (simo5) [inline src/endian.h:3]: I would rather use the usual ifdef based guards here

**Review by simo5 — CHANGES_REQUESTED** [review]
Boy is it ugly to support multiple platforms ... I wonder if this is the end of it, do we have specific uses in mind though?

That said, please move the file under a directory named "isc" like with the "oasis" headers.

File: src/Makefile.am (hunk 1 above)
Comment (oerdnj) [reply src/Makefile.am:8]: > Boy is it ugly to support multiple platforms ... I wonder if this is the end of it, do we have specific uses in mind though?

It compiles fine on macOS now.

> That said, please move the file under a directory named "isc" like with the "oasis" headers.

Why? There's nothing ISC specific, and I (personally) wrote the header, so it's ok to have it as public domain or whatever. There's no magic in there.

Comment (oerdnj) [reply src/Makefile.am:8]: > why this change?

This was unintentional. I do have some white space formatting macros in my emacs, so perhaps it reformatted the whitespace. I'll revert this.

File: src/endian.h (hunk 2 above)
Comment (oerdnj) [reply src/endian.h:3]: `#pragma once` is well supported even in ancient compilers, but ok

Comment (oerdnj) [reply src/endian.h:3]: Also with `#pragma once` you don't have to come up with fancy names like `_P11PROV_ENDIAN_H`.

File: src/store.c [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 #include "provider.h"
 #include <openssl/store.h>
-#include <endian.h>
+#include "endian.h"
```
Comment (simo5) [inline src/store.c:5]: Move endian.h -> isc/endian.h please

**Review by simo5 — CHANGES_REQUESTED** [review]

**Comment by simo5** [comment]
I assume this si a file generally provided by ISC ?
This is why I asked to move it under an isc/ directory.
If it is not something ISC regularly distributes (like the Oasis files) and I misunderstood, then I think I would still like to add it under a directory like "compat" or "platform" to indicate this is not a file to use to put in random new stuff but only compatibility layers for other platforms.

**Comment by oerdnj** [comment]
It's not something that we distribute. It's just I had to solve this for BIND 9, so platform/ works for me...

File: src/Makefile.am [hunk 4]
```diff
@@ -7,6 +7,7 @@ SHARED_EXT=@SHARED_EXT@
 pkcs11_provider_la_SOURCES = \
 	asymmetric_cipher.c \
 	debug.c \
+	endian.h \
```
Comment (simo5) [inline src/Makefile.am:10]: think this need to be changed accordingly

Comment (oerdnj) [reply src/Makefile.am]: Yep, good catch, fixed.

**Review by simo5 — APPROVED** [review]
